### ISSUE TITLE
cowsay: -l option not handled in perl-5.38.2

### ISSUE THREAD

**Issue by techie2000** [NixOS/nixpkgs#327961]
### Describe the bug
Executing `cowsay` with the `-l` option results `Use of uninitialized value $tail in join or string at /nix/store/b1n6f1pvkycj24warny93spccbnj9bx3-perl-5.38.2/lib/perl5/5.38.2/Text/Wrap.pm line 34.` (but does still seemingly list the available cows).

### Steps To Reproduce
1. nix-shell -p cowsay
2. cowsay -l
3. observe erroneous message on line 3 of output

### Expected behavior
A list of cows with no python related output.

### Screenshots
```console
[nix-shell:~/nixos]$ cowsay -l
Listing cowfiles in pretty format
Cow files in /nix/store/yqz388r6bjfkp3736inwzc5j4v9n4zba-cowsay-3.7.0/share/cowsay/site-cows:
Use of uninitialized value $tail in join or string at /nix/store/b1n6f1pvkycj24warny93spccbnj9bx3-perl-5.38.2/lib/perl5/5.38.2/Text/Wrap.pm line 34.


Cow files in /nix/store/yqz388r6bjfkp3736inwzc5j4v9n4zba-cowsay-3.7.0/share/cowsay/cows:
DragonAndCow Example Frogs MechAndCow Stegosaurus TextBalloon TuxStab
beavis.zen blowfish bong bud-frogs bunny cheese cower daemon default dragon
dragon-and-cow elephant elephant-in-snake eyes flaming-sheep ghostbusters
head-in hellokitty kiss kitty koala kosh llama luke-koala meow milk moofasa
moose mutilated ren satanic sheep skeleton small sodomized stegosaurus
stimpy supermilker surgery telebears three-eyes turkey turtle tux udder
vader vader-koala www
```

### Additional context
I have looked to see if there are any related issues or pull requests on the underlying project [https://github.com/cowsay-org/cowsay](https://github.com/cowsay-org/cowsay) but don't see anything reported there, i.e. an issue with the underlying release (3.7) rather than the nixos wrapper itself.

### Notify maintainers
@rbvermaa @anthonyroussel 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.36, NixOS, 24.05 (Uakari), 24.05.20240714.53e81e7`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.4`
 - nixpkgs: `/nix/store/hkv63lvykwr520xf8yv2gcw3s78fjfpd-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by anthonyroussel** [comment]
Indeed, looks like this warning happens because `share/cowsay/site-cows` is empty.

**Comment by panchoh** [comment]
> Indeed, looks like this warning happens because `share/cowsay/site-cows` is empty and has no parseable cow files.

This is part of a global effort to reduce methane emissions.

I know that today is not April Fools’ Day, but please understand that I __had__ to do it ;-P

**Comment by anthonyroussel** [comment]
Fixed by #337113 (https://github.com/cowsay-org/cowsay/releases/tag/v3.8.3)
